src/web/mjs/connectors/AstralLibrary.mjs

```diff
@@ -10,5 +10,28 @@ export default class AstralLibrary extends WordPressMadaraNovel {
         this.url = 'https://astrallibrary.net';
 
         this.novelObstaclesQuery = 'div.ad';
+        this.queryMangas = 'div.item-summary';
+    }
+
+    async _getMangasFromPage(page) {
+        let request = new Request(new URL('page/'+page+1,'https://www.astrallibrary.net/manga-tag/manga/'));
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element.querySelector('.list-chapter a'), request.url),
+                title: element.querySelector('a').textContent.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let chap = manga.id.match(/(\d+)\/$/)[1]
```

Comment:
Use the original `manga.id` and get the latest chapter number on the fly instead.
```javascript
const uri = new URL(this.url);
uri.searchParams.set('post_type', 'wp-manga');
uri.searchParams.set('s', manga.title);
const request = new Request(uri, this.requestOption);
const data = await fetchDOM(request, 'div.latest-chap span.chapter a');
// from this point follow the same drill by generating the chapters from the latest chapter number ...
```